versions/schematic-2.md

```diff
@@ -0,0 +1,196 @@
+<a name="metadataDate"></a>Date | `long` | The date that this schematic was created on. This is specified as milliseconds since the Unix epoch.
+<a name="metadataRequiredMods"></a>RequiredMods | `string`[] | An array of mod ids which have blocks which are referenced by this schematic's defined [Palette](#schematicPalette).
+
+##### Metadata Object Example:
+
+```js
+{
+    "Name": "My Schematic",
+    "Author": "Author Name",
+    "RequiredMods": [
+        "a_mod",
+        "another_mod"
+    ]
+}
+```
+
+#### <a name="paletteObject"></a>Palette Object
+
+An object which holds a mapping of a block state id to an index. The indices are recommended to start at `0` and may be no more than [`PaletteMax - 1`](#schematicPaletteMax). If the palette is not specified then the global mapping of ids is used instead. This is not recommended as it both expands the size of your blockdata and removes cross server and cross version compatibility for modded blocks which may change ids due to a variety of reasons.
+
+#### Block State Ids
+
+The format of the Block State identifier is the id of the block type and a set of comma-separated property `key=value` pairs surrounded by square brackets. If the block has no properties then they can be excluded. The block type id is specified as a [Resource Location](#defResourceLocation).
+
+For example the air block has no properties so its id representation would be just the block type id `minecraft:air`. The planks block however has an enum property for the `variant` so its id would be `minecraft:planks[variant=oak]`. Properties should be ordered with their keys in alphabetical ordering.
+
+##### Fields
+
+Field Pattern | Type | Description
+---|:---:|---
+<a name="paletteEntry"></a>{blockstate} | `integer` | A single entry mapping a blockstate to an index.
+
+##### Palette Object Example
+
+```js
+"Palette" {
+    "minecraft:air": 0,
+    "minecraft:planks[variant=oak]": 1,
+    "a_mod:custom": 2
+}
+```
+
+#### <a name="tileEntityObject"></a>Tile Entity Object
+
+An object to specify a tile entity which is within the region. Tile entities are used by Minecraft to store additional data for a block (such as the lines of text on a sign). The fields used to describe a tile entity vary for each type, however the structure will be the same as used by the [Minecraft Chunk Format](http://minecraft.gamepedia.com/Chunk_format#Block_entity_format).
+
+##### Fields
+
+Field Pattern | Type | Description
+---|:---:|---
+<a name="tileEntityPos"></a>Pos | `integer`[3] | **Required.** The position of the tile entity relative to the `[0, 0, 0]` position of the schematic (without the [offset](#schematicOffset) applied). Must contain exactly 3 integer values.
+<a name="tileEntityId"></a>Id | `string` | **Required.** The id of the tile entity type defined by this Tile Entity Object, specified as a [Resource Location](#defResourceLocation). This should be used to identify which fields should be required for the definition of this type.
+
```

Comment:
maybe worth adding `{extra}` here like the entity fields for consistency/clarity.